lib/ide/src/controller/module.rs

```diff
@@ -0,0 +1,64 @@
+//! Module Controller.
+//!
+//! The module controller keeps cached module state (module state is AST+Metadata or equivalent),
+//! and uses it for synchronizing state for text and graph representations. It provides method
+//! for registering text and graph changes. If for example text represntation will be changed, there
+//! will be notifications for both text change and graph change.
+//!
+//! This module is still on WIP state, for now it contains stubs only.
+
+use crate::prelude::*;
+
+use shapely::shared;
+
+
+/// Structure uniquely identifying module location in the project.
+/// Mappable to filesystem path.
+#[derive(Clone,Debug,Eq,Hash,PartialEq)]
+pub struct Location(pub String);
+
+impl Location {
+    /// Obtains path (within a project context) to the file with this module.
+    pub fn to_path(&self) -> file_manager_client::Path {
+        // TODO [mwu] Extremely provisional. When multiple files support is
+        //            added, needs to be fixed, if not earlier.
+        let Location(string) = self;
+        let result = format!("./{}.luna", string);
+        file_manager_client::Path::new(result)
+    }
+}
+
+shared! { ControllerHandle
```

Comment:
So you assumes, that we are using full module paths everywhere. And I have a problem with something like `Location` - for it the best use would be `module::Location` But when I import `module` then someone could use `module::Handle` and this for potential reader is unclear, because he doesn't see that this is controller.

So finally, I think that treating modules as a part of structure name is impractical, and I'll rename it just to `ModuleController`

Reply:
Well. I changed my mind again and did as you please, now everywhere we use just `Handle`